nemo_skills/pipeline/utils/ray_backend.py

```
@@ -0,0 +1,820 @@
+            job_log_file: str | None,
+            stage_run_id: str | None,
+            stage_manifest_file: str | None,
+            submission_id: str | None,
+        ) -> Dict[str, Any]:
+            """Poll a single Ray Job until it reaches a terminal state."""
+            last_logs = ""
+            while True:
+                status_str = self._to_status_str(client.get_job_status(job_id))
+                try:
+                    logs = client.get_job_logs(job_id) or ""
+                except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)